libqtile/state.py

```diff
@@ -42,17 +42,16 @@ def __init__(self, qtile):
             self.layout_map[group.name] = {}
             for layout in group.layouts:
                 self.layout_map[group.name][layout.name] = layout
-                layout.group = None
-                if isinstance(layout, libqtile.layout.tree.TreeTab):
+                layout.group = None                 # We Need this for serialization to work.
+                if isinstance(layout, libqtile.layout.tree.TreeTab):  # Special cases as they have layout objects within there structure.
                         layout._draw = True if layout._panel is not None else False
                         layout._drawer = None
                         layout._panel = None
                         layout._layout = None
                 if isinstance(layout, libqtile.layout.slice.Slice):
-                    layout.fallback.group = None
+                    layout.fallback.group = None # to make them pickelable
```

Comment:
Be sure you have 2 spaces before the `#` for in-line comments, you have some flake8 failures